Assets/Scripts/Weapons/Weapon.cs

```diff
@@ -0,0 +1,50 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Weapon : MonoBehaviour
+{
+    Dictionary<string, bool> _firingBehavior = new Dictionary<string, bool>()
+    {
+        { "singleShot", true }, // singleShot == !repeater
+        { "shotgun", false },
+        { "repeater", false }
+    };
+
+    IHitBehaviour _hitBehaviour;
+    float _fireRate; // shots per second
+    int _magazineSize;
+    int _currentMagazine;
+    double lastShot = 0;
+
+    void Start()
+    {
+        _hitBehaviour = new BulletBehaviour();
```

Comment:
hitBehaviour should be set from a parameter, to not lock in Bullets. This is fine to leave in though, since we won't be using different attack behaviours until Melee Tank.